File: library_samples/Python3/ocs_sample_library_preview/Dataview/FieldSource.py
from enum import Enum


class FieldSource(Enum):
    """
    enum 0-6  fully inclusive
    """
    NotApplicable = 0
    Id = 1
    Name = 2
    PropertyId = 3
    PropertyName = 4
    Tags = 5
    Metadata = 6

File: library_samples/Java/ocs_sample_library_preview/src/main/java/com/github/osisoft/ocs_sample_library_preview/dataviews/FieldSource.java
package com.github.osisoft.ocs_sample_library_preview.dataviews;

public enum FieldSource {

    NotApplicable("NotApplicable"), Id("Id"), Name("Name"), PropertyId("PropertyId"), PropertyName("PropertyName"),
    Tags("Tags"), Metadata("Metadata");

    private final String FieldSource;

    private FieldSource(String id) {
        this.FieldSource = id;
    }

    public String getValue() {
        return FieldSource;
    }
}
